kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,746 @@
+        Parameters linear_solver_parameters(R"({
+            "solver_type": "cg_solver",
+            "tolerance": 1.0e-8,
+            "max_iteration": 200,
+            "preconditioner_type": "none",
+            "scaling": false
+        })");
+        auto p_linear_solver = Kratos::make_shared<CGSolver<TSparseSpace, TDenseSpace>>(linear_solver_parameters);
+
+        // Create the linear strategy
+        SchemePointerType p_scheme = Kratos::make_shared<ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace, TDenseSpace>>();
+
+        bool calculate_norm_dx = false;
+        bool calculate_reactions = false;
+        bool reform_dof_at_each_iteration = false;
+        BuilderSolverPointerType p_builder_and_solver = Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>>(p_linear_solver);
+
+        Model &current_model = mrBaseModelPart.GetModel();
+        ModelPart &r_aux_model_part = current_model.GetModelPart(mAuxModelPartName);
+
+        mpSolvingStrategy = Kratos::make_unique<ResidualBasedLinearStrategy<TSparseSpace, TDenseSpace, TLinearSolver>>(
+            r_aux_model_part,
+            p_scheme,
+            p_linear_solver,
+            p_builder_and_solver,
+            calculate_reactions,
+            reform_dof_at_each_iteration,
+            calculate_norm_dx);
+
+        mpSolvingStrategy->Check();
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void CalculateIntersections()
+    {
+        mpFindIntersectedGeometricalObjectsProcess = Kratos::make_shared<FindIntersectedGeometricalObjectsProcess>(mrBaseModelPart, mrSkinModelPart);
+        mpFindIntersectedGeometricalObjectsProcess->Initialize();
+        mpFindIntersectedGeometricalObjectsProcess->FindIntersections();
+    }
+
+    void ClearIntersections()
+    {
+        mpFindIntersectedGeometricalObjectsProcess->Clear();
+    }
+
+    const Vector SetDistancesVector(ModelPart::ElementIterator ItElem)
+    {
+        auto &r_geom = ItElem->GetGeometry();
+        Vector nodal_distances(r_geom.PointsNumber());
+
+        if (mLevelSetType == "continuous"){
+            // Continuous nodal distance function case
+            for (unsigned int i_node = 0; i_node < r_geom.PointsNumber(); ++i_node) {
+                nodal_distances[i_node] = r_geom[i_node].FastGetSolutionStepValue(DISTANCE);
+            }
+        } else if (mLevelSetType == "discontinuous") {
+            // Discontinuous elemental distance function case
+            nodal_distances = ItElem->GetValue(ELEMENTAL_DISTANCES);
```

Comment:
are you checking at some point if this is initialized?

Reply:
A `SetValue(ELEMENTAL_DISTANCES)` is always done in the `Initialize()` method of the discontinuous distance calculation method.